worker.go

```diff
@@ -31,41 +31,54 @@ type worker struct {
 var (
 	workers          []*worker
 	watcherIsEnabled bool
+	startupFailChan  chan (error)
 )
 
 func initWorkers(opt []workerOpt) error {
 	workers = make([]*worker, 0, len(opt))
-	workersReady := sync.WaitGroup{}
 	directoriesToWatch := getDirectoriesToWatch(opt)
 	watcherIsEnabled = len(directoriesToWatch) > 0
+	totalThreadsToStart := 0
 
 	for _, o := range opt {
 		w, err := newWorker(o)
 		if err != nil {
 			return err
 		}
+		totalThreadsToStart += w.num
 		workers = append(workers, w)
 	}
 
+	startupFailChan = make(chan error, totalThreadsToStart)
+	workersReady := sync.WaitGroup{}
+	workersReady.Add(totalThreadsToStart)
```

Comment:
You could use the new `WaitGroup.Go()` function here: https://pkg.go.dev/sync#WaitGroup.Go